packages/react-native/src/InAppMessaging/components/BannerMessage/__tests__/BannerMessage.spec.tsx

```diff
@@ -83,39 +81,60 @@ describe('BannerMessage', () => {
         expectedProps: { children: 'body content' },
       },
     ],
+  ])(
+    'correctly handles a %s prop',
+    (key, testID, { testProps, expectedProps }) => {
+      mockUseMessageImage.mockReturnValueOnce({
+        hasRenderableImage: false,
+        imageDimensions: { height: undefined, width: undefined },
+        isImageFetching: false,
+      });
+
+      const props = { ...baseProps, [key]: testProps };
+
+      const { getByTestId } = render(<BannerMessage {...props} />);
+      const testElement = getByTestId(testID);
+
+      for (const [key, value] of Object.entries(expectedProps)) {
+        expect(testElement.props[key]).toEqual(value);
+      }
+    }
+  );
+
+  it.each([
     [
       'primaryButton',
       IN_APP_MESSAGING_TEST_ID.PRIMARY_BUTTON,
       {
         testProps: { onAction, title: 'primary button' },
-        expectedProps: { children: 'primary button', onPress: onAction },
       },
     ],
     [
       'secondaryButton',
       IN_APP_MESSAGING_TEST_ID.SECONDARY_BUTTON,
       {
         testProps: { onAction, title: 'secondary button' },
-        expectedProps: { children: 'secondary button', onPress: onAction },
       },
     ],
-  ])(
-    'correctly handles a %s prop',
-    (key, testID, { testProps, expectedProps }) => {
-      mockUseMessageImage.mockReturnValueOnce({
-        hasRenderableImage: false,
-        imageDimensions: { height: undefined, width: undefined },
-        isImageFetching: false,
-      });
+  ])('correctly handles a %s prop', (key, testID, { testProps }) => {
```

Comment:
Not sure how this got past linting since it we have a rule preventing reuse of titles (`jest/no-identical-title`, probably because the title is computed), but this could be confusing for future maintainers since we are using the same title template on line 85.